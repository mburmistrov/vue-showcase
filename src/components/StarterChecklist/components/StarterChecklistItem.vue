<template lang="pug">
.starter-checklist-item(:class="classes")
  .starter-checklist-item__content(v-html="$t(`starterChecklist.items.${itemKey}`)")
</template>
<script>
export default {
  props: {
    itemKey: {
      type: String,
      required: true
    },
    isCompleted: {
      type: Boolean,
      required: true
    }
  },
  computed: {
    classes() {
      return {
        'starter-checklist-item--completed': this.isCompleted
      }
    }
  }
}
</script>
<style lang="sass">
@import "~/src/styles/icons/mixins"
=item-icon($name)
  +md-icon($name)
  font-size: 25px
  margin-right: 10px
  position: relative
  top: -3px
.starter-checklist-item
  $text-color: #737376
  display: flex
  padding: 6px 0
  font-size: 16px
  font-weight: 400
  &:before
    +item-icon(circle-o)
    color: #ccc
  .starter-checklist-item__content
    color: $text-color
  &.starter-checklist-item--completed
    &:before
      +item-icon(check-circle)
      color: #00A591
    .starter-checklist-item__content
      color: transparentize($text-color, .55)
      text-decoration: line-through
</style>
