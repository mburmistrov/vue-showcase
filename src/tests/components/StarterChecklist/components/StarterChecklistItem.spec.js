import Vue from 'vue'
import { shallowMount } from '@vue/test-utils'
import StarterChecklistItem from '@/components/StarterChecklist/components/StarterChecklistItem'
import { merge } from 'lodash'

describe('StarterChecklistItem', () => {
  function createWrapper(overrides) {
    const defaultMountingOptions = {
      propsData: {
        itemKey: 'itemKey',
        isCompleted: false
      }
    }
    return shallowMount(StarterChecklistItem, merge(defaultMountingOptions, overrides))
  }

  describe('template', () => {
    function getL10nKeyForItemKey(itemKey) {
      return `starterChecklist.items.${itemKey}`
    }

    it('should have completed class if isCompleted prop is true', async () => {
      const completedClass = 'starter-checklist-item--completed'
      const wrapper = createWrapper()
      expect(wrapper.classes()).not.toContain(completedClass)
      wrapper.setProps({ isCompleted: true })
      await Vue.nextTick()
      expect(wrapper.classes()).toContain(completedClass)
    })

    it('should render content from items l10n key', async () => {
      const firstItemKey = 'firstKey'
      const secondItemKey = 'secondKey'
      const wrapper = createWrapper({ propsData: { itemKey: firstItemKey } })
      expect(wrapper.text()).toContain(getL10nKeyForItemKey(firstItemKey))
      wrapper.setProps({ itemKey: secondItemKey })
      await Vue.nextTick()
      expect(wrapper.text()).toContain(getL10nKeyForItemKey(secondItemKey))
    })
  })

  describe('snapshots', () => {
    it('should match snapshot', () => {
      const wrapper = createWrapper()
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
