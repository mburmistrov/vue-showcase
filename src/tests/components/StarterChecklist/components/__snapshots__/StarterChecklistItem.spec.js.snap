// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarterChecklistItem snapshots should match snapshot 1`] = `
<div
  class="starter-checklist-item"
>
  <div
    class="starter-checklist-item__content"
  >
    starterChecklist.items.itemKey
  </div>
</div>
`;
